js/ui/overlay/ui.overlay.js

```diff
@@ -488,6 +489,11 @@ const Overlay = Widget.inherit({
     _normalizePosition: function() {
         const position = this.option('position');
         this._position = typeof position === 'function' ? position() : position;
+        const container = this.option('container');
+        if(container && !isWindow(container)) {
+            this._position = this._position || {};
+            this._position.container = getAvailableBoundsContainer(container);
```

Comment:
mb we should name it like `boundaryContainer` to avoid confusion boundary container and real container in which overlay is positioned?

Reply:
I think this naming can add confusion with the boundary in the 'calculatePosition' function